src/main/java/fi/aalto/cs/apluscourses/model/Exercise.java

```diff
@@ -132,6 +137,14 @@ public int getMaxSubmissions() {
     return maxSubmissions;
   }
 
+  public double getLatePenalty() {
+    return latePenalty;
+  }
+
+  public boolean isLate() {
+    return this.latePenalty != 0.0;
+  }
```

Comment:
It's a bit weird that an exercise would have a late penalty, since it's not exercises that are late or penalized but submissions. I then saw that submission results also have a late penalty, I wonder why that information is duplicated?

Reply:
Would a better way be to store the best submission and check the penalty of that